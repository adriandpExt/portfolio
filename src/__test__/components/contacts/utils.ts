import { fireEvent } from "@testing-library/react";

export const changeInputValue = (
  container: HTMLElement,
  testId: string,
  value: string
) => {
  const inputElement = container.querySelector(
    `[data-testid="${testId}"] input`
  );
  if (!inputElement) {
    return new Error(`Input element with data-testid "${testId}" not found`);
  }
  fireEvent.change(inputElement, { target: { value } });
};

export const changeTextareaValue = (
  container: HTMLElement,
  testId: string,
  value: string
) => {
  const inputElement = container.querySelector(
    `[data-testid="${testId}"] textarea`
  );
  if (!inputElement) {
    return new Error(`Input element with data-testid "${testId}" not found`);
  }
  fireEvent.change(inputElement, { target: { value } });
};
