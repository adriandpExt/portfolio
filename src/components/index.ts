export * from "./appbar";
export * from "./svg-icons";
export * from "./aside";
