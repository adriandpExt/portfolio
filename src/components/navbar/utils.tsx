import AssignmentIndIcon from "@mui/icons-material/AssignmentInd";
import HomeIcon from "@mui/icons-material/Home";
import AppsIcon from "@mui/icons-material/Apps";
import ContactMailIcon from "@mui/icons-material/ContactMail";
import InfoIcon from "@mui/icons-material/Info";

export const menuItems = [
  { listIcon: <HomeIcon />, listText: "Home", listPath: "/portfolio " },
  {
    listIcon: <AssignmentIndIcon />,
    listText: "Resume",
    listPath: "/portfolio/cv",
  },
  {
    listIcon: <InfoIcon />,
    listText: "About",
    listPath: "/portfolio/about",
  },
  {
    listIcon: <AppsIcon />,
    listText: "Projects",
    listPath: "/portfolio/project",
  },
  {
    listIcon: <ContactMailIcon />,
    listText: "Contact",
    listPath: "/portfolio/contact",
  },
];
