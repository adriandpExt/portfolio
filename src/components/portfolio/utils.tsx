// import avatar from "../../assets/image/avatar.png";
import empf1 from "../../assets/empf/empf1.png";
import empf2 from "../../assets/empf/empf2.png";
import empf3 from "../../assets/empf/empf3.png";

export const projects = [
  {
    name: "EMPF",
    description: `EMPF is a centralised electronic platform that aims to standardise and automate the administration processes of MPF schemes. An electronic platform to be launched by the Mandatory Provident Fund Schemes Authority (MPFA), it will cover all 12 MPF trustees, as well as the 357,000 employers and 4.7 million members that Hong Kong's mandatory pension scheme covers. Collaboration with other developer`,
    image: [empf1, empf2, empf3],
  },
];
