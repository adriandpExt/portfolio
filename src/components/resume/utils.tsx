export const details = [
  {
    jobtitle: "Solutions Developer",
    companyName: "LENOVO PCCW SOLUTIONS PHILIPPINES INC",
    year: "2022-2024",
    description: `Motivated Solution Developer with a dedication to delivering sophisticated and effective web applications. I am in search of a challenging position where I can utilize my two years of expertise in front-end development and UI design, employing contemporary technologies to contribute to the company's success`,
  },
];
