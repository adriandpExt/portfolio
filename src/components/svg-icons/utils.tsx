export type IconName =
  | "ic_gmail"
  | "ic_mobile"
  | "ic_facebook"
  | "ic_github"
  | "ic_linkedin"
  | "ic_vue"
  | "ic_react"
  | "ic_html"
  | "ic_css"
  | "ic_javascript"
  | "ic_bootstrap"
  | "ic_material_ui"
  | "ic_tailwind";
