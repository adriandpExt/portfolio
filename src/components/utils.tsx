export const toggleDarkTheme = (isDarkMode: boolean) => {
  return isDarkMode ? "bg-gray-900 text-white" : "bg-gray-200 text-gray-800";
};

export const toggleLightTheme = (isDarkMode: boolean) => {
  return isDarkMode ? "bg-gray-200 text-gray-800" : "bg-gray-900 text-white";
};

export const handleToggleDropIcon = (isDarkMode: boolean) => {
  return isDarkMode ? "bg-gray-200  text-gray-800" : "bg-gray-900  text-white";
};
