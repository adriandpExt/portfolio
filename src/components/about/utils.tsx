import { IconName } from "../svg-icons/utils";

export const techStack: IconName[] = [
  "ic_react",
  "ic_vue",
  "ic_javascript",
  "ic_typescript",
  "ic_html",
  "ic_css",
  "ic_bootstrap",
  "ic_tailwind",
  "ic_material_ui",
];
export const techTools: IconName[] = ["ic_git", "ic_mysql", "ic_vscode"];
