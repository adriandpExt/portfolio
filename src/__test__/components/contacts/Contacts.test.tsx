import "@testing-library/jest-dom";
import { fireEvent, render, waitFor } from "@testing-library/react";

import { Contacts } from "../../../components";
import { changeInputValue, changeTextareaValue } from "./utils.ts";

jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
}));

describe("Contact Component Tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("Render Test", () => {
    const { container } = render(<Contacts />);

    expect(container).toMatchSnapshot();
  });

  test("renders the contact form correctly", () => {
    const { getByTestId, getByLabelText, getByText } = render(<Contacts />);

    expect(getByText("Contact Me")).toBeInTheDocument();

    expect(getByTestId("fullname")).toBeInTheDocument();
    expect(getByTestId("email")).toBeInTheDocument();
    expect(getByTestId("message")).toBeInTheDocument();

    expect(getByLabelText("Fullname")).toBeInTheDocument();
    expect(getByLabelText("Email")).toBeInTheDocument();
    expect(getByLabelText("Message")).toBeInTheDocument();
  });

  test("submit form with valid data", async () => {
    const { getByTestId, getByText, container } = render(<Contacts />);

    changeInputValue(container, "fullname", "John Doe");
    changeInputValue(container, "email", "john@example.com");
    changeTextareaValue(container, "message", "Test message");

    fireEvent.submit(getByTestId("contact-form"));

    await waitFor(() => {
      expect(getByText("Contact Me")).toBeInTheDocument();
    });
  });
});
