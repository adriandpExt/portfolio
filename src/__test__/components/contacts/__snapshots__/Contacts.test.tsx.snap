// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component Tests Render Test 1`] = `
<div>
  <div
    class="MuiBox-root css-a2eb3v"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1v470xt"
      >
        <form
          data-testid="contact-form"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1f0uwg7-MuiTypography-root"
          >
            Hire or Contact me...
          </h5>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1d5kksk-MuiFormControl-root-MuiTextField-root"
            data-testid="fullname"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Fullname
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                name="fullname"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Fullname
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1f85017-MuiFormControl-root-MuiTextField-root"
            data-testid="email"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r1:"
              id=":r1:-label"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                name="email"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1d5kksk-MuiFormControl-root-MuiTextField-root"
            data-testid="message"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              Message
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                name="message"
                rows="4"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Message
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-xj0r1l-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Contact Me
            <span
              class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SendIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiBottomNavigation-root css-1rj3k27-MuiBottomNavigation-root"
        >
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-3yflfb-MuiButtonBase-root-MuiBottomNavigationAction-root"
            data-testid="facebook"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FacebookIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1gh6k7f-MuiBottomNavigationAction-label"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-3yflfb-MuiButtonBase-root-MuiBottomNavigationAction-root"
            data-testid="linkedin"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LinkedInIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1gh6k7f-MuiBottomNavigationAction-label"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-3yflfb-MuiButtonBase-root-MuiBottomNavigationAction-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EmailIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1gh6k7f-MuiBottomNavigationAction-label"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-3yflfb-MuiButtonBase-root-MuiBottomNavigationAction-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CallIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56-.35-.12-.74-.03-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1gh6k7f-MuiBottomNavigationAction-label"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
